onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -1381,6 +1379,37 @@ TEST_F(GraphTransformationTests, ReshapeFusionConcatSubgraph) {
   }
 }
 
+TEST_F(GraphTransformationTests, ReshapeFusionConcatSubgraphNotTriggered) {
+  auto model_uri = MODEL_FOLDER "fusion/reshape_fusion_concat_subgraph_not_triggered.onnx";
+  std::shared_ptr<Model> p_model;
+  ASSERT_TRUE(Model::Load(model_uri, p_model, nullptr, *logger_).IsOK());
+  Graph& graph = p_model->MainGraph();
+
+  onnxruntime::GraphTransformerManager graph_transformation_mgr{5};
+  graph_transformation_mgr.Register(onnxruntime::make_unique<ReshapeFusion>(), TransformerLevel::Level1);
+  auto ret = graph_transformation_mgr.ApplyTransformers(graph, TransformerLevel::Level1, *logger_);
+  ASSERT_TRUE(ret.IsOK());
+
+  // None of the subgraphs leading to the Concat node will not trigger the optimization
+  // as an additional pad value of 1 is inserted thus making the input to the Concat
+  // [1, 10], [1, 20], and [1, 30]
+
```

Comment:
Would be helpful to add info on which check in the optimizer fails due to the additional pad value.

Reply:
Not sure is this is needed anymore after we remove the additional Pads. Two branches match the pattern and one won't  (the branch with the Pad) and hence won't trigger the optimization (which we test for). So do we really need to check which branch doesn't trigger the optimization ?